Python/optimizer_bytecodes.c

```diff
@@ -448,8 +456,10 @@ dummy_func(void) {
         top = bottom;
     }
 
-    op(_SWAP, (bottom, unused[oparg-2], top --
-        top, unused[oparg-2], bottom)) {
+    op(_SWAP, (bottom_in, unused[oparg-2], top_in --
+        top_out, unused[oparg-2], bottom_out)) {
+        bottom_out = bottom_in;
+        top_out = top_in;
```

Comment:
Hm, this makes me a bit sad. We can't reuse inputs as outputs anymore?

Not a big deal if it's challenging to get right, I'm just a bit surprised.

Reply:
The model is that all inputs are taken from the stack, then all outputs are pushed to the stack. We leave any values on the stack that don't move.

We could handle this in the code generator, by doing the rewrite automatically, but I'm not sure it is worth the effort for the 2 or 3 instructions that would be simplified.